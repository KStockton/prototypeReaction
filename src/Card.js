import React, { Component } from 'react'
import './Card.scss'
import QuestionTitle from './QuestionTitle'
import './images/jigsawjpeg.jpeg'

export default class Card extends Component {

 render() {
    let displayInfo
   if(this.props.topicChoice) {
    displayInfo = <QuestionTitle topicChoice={this.props.topicChoice}
         />
      } else {
          displayInfo = <article>
                        <h2>Would you like to play a game?</h2>
                        <h5>You have 30 seconds to test your knowledge on string and array prototypes</h5>
                        <p>Select a Prototype to Above to Begin</p>
                      </article> 
        } 
  return (
   <div className="Card-wrapper">
   <section className="title-wrapper">
    {displayInfo}
   </section>
   </div>
  )
 }
}