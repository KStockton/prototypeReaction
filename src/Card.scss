.Card-wrapper{
  display: flex;
  justify-content: center;
}

.title-wrapper {
  background-color: #ffffff;
  width: 861px;
  height: 384px;
  border-radius: 8px;
}