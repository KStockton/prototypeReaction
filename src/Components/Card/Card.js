import React, { Component } from 'react';
import QuestionTitle from '../QuestionTitle/QuestionTitle';
export default class Card extends Component {

  render() {
    const { topicChoice } = this.props;
    let displayInfo;

    if (topicChoice) {
      displayInfo = <QuestionTitle topicChoice={topicChoice}/>;
    } else {
      displayInfo = <article>
        <h2>Introduction</h2>
        <p>Test your knowledge on string and array prototypes.</p>
        <p>Begin By Selecting Array or String Prototypes Above to Begin</p>
      </article>; 
    } 
    return (
      <div className="Card-wrapper">
        <section className="title-wrapper">
          {displayInfo}
        </section>
      </div>
    );
  }
}