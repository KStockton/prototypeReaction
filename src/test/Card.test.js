import React from 'react';
import Card from '../Card';
import { shallow } from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() })

const mocktopicChoice = [
  {
   id: "2932-tq-iwpjt-0oodks",
   category: "Array Prototypes",
   questions: "What does .unshift do to an array ?",
   answers: [
   "adds items to the beginning",
   "removes items from beginning",
   "adds items to the end"
   ],
   correctAnswer: "adds items to the beginning",
   correct: false,
   resource: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift"
  },
  {
   id: "2932-f5-o4cm9-lcfdks",
   category: "Array Prototypes",
   questions: "What array prototype checks if an array has a certain value and returns true or false ?",
   answers: [
   ".findIndex",
   ".lastIndexOf",
   ".includes"
   ],
   correctAnswer: ".includes",
   correct: false,
   resource: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes"
  }
]


describe('Card', () => {
  let wrapper;

  beforeEach(() => {
   wrapper = shallow(<Card />
   )
  });

  it('should match the snapshot with all data passed in', () =>{
   expect(wrapper).toMatchSnapshot()
  })
})
  describe('Card renders correctly when props are passed in', () =>{
    
    it('Card renders differently when props are passed in', () => {
      const wrapper = shallow(<Card topicChoice={mocktopicChoice} />)
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render one QuestionTitle when props are true', () =>{
      const wrapper = shallow(<Card topicChoice={mocktopicChoice} />)
      const questionComponent = wrapper.find('QuestionTitle')
      expect(questionComponent.length).toEqual(1)
    })
  })
