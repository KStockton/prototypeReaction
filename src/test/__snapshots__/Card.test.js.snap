// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly when props are passed in Card renders differently when props are passed in 1`] = `
<div
  className="Card-wrapper"
>
  <section
    className="title-wrapper"
  >
    <QuestionTitle
      topicChoice={
        Array [
          Object {
            "answers": Array [
              "adds items to the beginning",
              "removes items from beginning",
              "adds items to the end",
            ],
            "category": "Array Prototypes",
            "correct": false,
            "correctAnswer": "adds items to the beginning",
            "id": "2932-tq-iwpjt-0oodks",
            "questions": "What does .unshift do to an array ?",
            "resource": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift",
          },
          Object {
            "answers": Array [
              ".findIndex",
              ".lastIndexOf",
              ".includes",
            ],
            "category": "Array Prototypes",
            "correct": false,
            "correctAnswer": ".includes",
            "id": "2932-f5-o4cm9-lcfdks",
            "questions": "What array prototype checks if an array has a certain value and returns true or false ?",
            "resource": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes",
          },
        ]
      }
    />
  </section>
</div>
`;

exports[`Card should match the snapshot with all data passed in 1`] = `
<div
  className="Card-wrapper"
>
  <section
    className="title-wrapper"
  >
    <article>
      <h2>
        Knowledge Is Power. Let's Begin!
      </h2>
      <p>
        Test your knowledge on string and array prototypes.
      </p>
      <p>
        Select Option Above to Begin
      </p>
    </article>
  </section>
</div>
`;
